app/controllers/api/v2/courses_controller.rb

```diff
@@ -133,9 +133,12 @@ def create
         create_new_course
       end
 
-      def send_vacancies_filled_notification
+      def send_vacancies_updated_notification
```

Comment:
why is this endpoint needed? Should publish not just update the course and the api works out if a notification should be sent? Why does the client itself need to know state and then decide if it should send a notification or not by hitting this endpoint?

Reply:
I'm going to get @defong and @SamJamCul to weigh in on this as they may have some more historical context. 

Whether a course actually has vacancies or not is actually determined by the SiteStatus resource - https://github.com/DFE-Digital/publish-teacher-training/pull/1242/files#diff-30720a018bf9d1c8a2a6274262bfb656R66. 

One option would be to create an endpoint for modifying a collection of SiteStatuses, else we need to check if a course has vacancies every time a single site status is updated.

Reply:
So one option I considered is making this a callback so when a `SiteStatus` is update it pings the course to see if it should send the notification. This would keep things API side but i think this will have race conditions depending on the order the site statuses get updated.

So could at the very least not keep the logic api side. eg
```ruby
vacancies_filled: params[:course][:vacancies_filled]
```

Why does the client (publish) need to tell the api if the vacancies of a course has been filled? The API should know this surely and determine this and not the client? 

It looks like the data has been shipped to publish, publish does this determination, then sends it back to the API.

Reply:
But yes I think the best solution would be to have an endpoint that can update a collection of resources in a transaction as you mentioned

Reply:
Ya, we've tried this before, and it's been a world of hurt. JSONAPI doesn't support this, you have to update resources individually. You _can_ update what the collection of resources that are associated to another resource, but that doesn't update the resources in the collection themselves, that just updates the association.

I suggest that we stick with updating resources individually and triggering a notification from publish, I don't think that that's too terrible given the limitations we're working with, but I do think that this adds ammunition to the argument that we should just merge publish and api, which I think suddenly makes more sense now that we know better where Find is going, etc.